src/parser/mod.rs

```diff
@@ -6210,6 +6210,20 @@ impl<'a> Parser<'a> {
         } else {
             let name = self.parse_object_name()?;
 
+            let partitions = if dialect_of!(self is MySqlDialect)
```

Comment:
Generally this crate also allows all syntax for the `GenericDialect`, such as

https://github.com/sqlparser-rs/sqlparser-rs/blob/4c3a4ad5a87ef8b4dd24a6e8540243e14fbfa148/src/parser/mod.rs#L756